packages/block-library/src/verse/test/edit.native.js

```diff
@@ -64,13 +64,15 @@ describe( 'Verse block', () => {
 		const verseTextInput = await screen.findByPlaceholderText(
 			'Write verse…'
 		);
-		typeInRichText( verseTextInput, 'A great statement.' );
+		typeInRichText( verseTextInput, 'A great statement.Again', {
+			finalSelectionStart: 18,
+			finalSelectionEnd: 18,
+		} );
```

Comment:
I've adjusted the tests. Took inspiration for Pullquote tests, where citation also creates a new default block.

https://github.com/WordPress/gutenberg/blob/8aa016e6445e3f1e909d2d900280b2ea96b353be/packages/block-library/src/pullquote/test/edit.native.js#L48-L56

Reply:
Given that the Verse block is for content like poetry and lyrics, line breaks should be expected.

> There is no ability to simultaneously press Shift and Return on most mobile OS virtual keyboards.

I agree that moving the cursor position is too cumbersome to be the only way to create a line break within a Verse block on mobile. As a possible alternative, we could consider a [soft return](https://en.wikipedia.org/wiki/Line_wrap_and_word_wrap), where pressing the shift key and then pressing enter on mobile creates a new line within the Verse block. Soft returns are a common pattern in multiline text input on mobile (like messaging apps) that may already be intuitive to users. A hard return (the behavior implemented in this PR) could create a new block, which would minimize divergence between web and mobile behavior.

Reply:
Thanks for the great feedback. @stokesman had the same concerns. It's clear to me now that users expect line breaks when pressing <kbd>Enter</kbd> or <kbd>Return</kbd> in the Verse block.

It would be easier to revert the changes and mark #52773 as `wontfix` for now.

> Should Preformatted behave like a "container" block, or should it align with Verse, Header, Paragraph? From my testing, Preformatted currently behaves like the former (as Verse did prior to this PR).

The container blocks support this feature differently when two consecutive <kbd>Enter</kbd>s create a new block. Maybe we can adopt similar logic for some non-container blocks as well.

Cc @annezazu, since you extracted the original issue from the GitHub discussions.

Reply:
If reverting this change on the web is an option, I'd like to add support for that, as I agree with sentiments already shared that users will be expecting/wanting line breaks in the Verse block.

> The container blocks support this feature differently when two consecutive Enters create a new block. Maybe we can adopt similar logic for some non-container blocks as well.

I'd also be wary of adding this as a solution as folks may want the freedom of adding blank lines between content in multiline blocks. In the case of the Verse block, for example, a poem may include multiple stanzas that should be separated with a blank line.

I'm afraid I don't have any bright ideas for solving the original problem of more easily exiting the Verse block. Perhaps after hitting return twice, the toolbar could emerge, so that the option of adding a new block is more readily available if users want to take it? Or perhaps there's some other creative approach that could be taken? 🤔 It would be interesting to figure out whether this is a common issue that those using multiline blocks face.

Reply:
> Perhaps after hitting return twice

That's exactly what I have would expected. Three times actually.



1. Writing: I am writing a verse `Enter`
2. `Still in the verse` _Thinking: I want a blank paragraph in between._ `Enter` 
3. `Still in the verse` Writing: I write the last paragraph in the verse.`Enter`
4. `Still in the verse` _Thinking: Blank paragraph, as above_  `Enter`
5. `Still in the verse` _Thinking: Okay, if I type now it'd be the same as step 3. But I don't type, so let's get out of here._ `Enter` 
7. `Out of the verse`

Reply:
To put it simply: It's not possible to create two blank paragraphs after each other.

Reply:
Ah, I see that @jameskoster suggested the same below and it can' t be done right now. 
However, this should be definitely be done in the future. Is anyone aware of a ticket, or should we utilise this one: https://github.com/WordPress/gutenberg/issues/52773 @Mamaduka Could you please reopen it again?